include/dlaf/factorization/qr/t_factor_mc.h

```diff
@@ -62,6 +90,157 @@ struct QR_Tfactor<Backend::MC, Device::CPU, T> {
+      TileElementSize va_size{vi.size().rows() - first_element_in_tile, x0.col()};
+      TileElementIndex va_start{first_element_in_tile, 0};
+      TileElementIndex vb_start{first_element_in_tile, x0.col()};
+
+      if (is_v0) {
+        t(x0) = tau;
+
+        // use implicit 1 for the 2nd operand
+        for (SizeType r = 0; !va_size.isEmpty() && r < va_size.cols(); ++r) {
+          const TileElementIndex i_v{va_start.row(), r + va_start.col()};
+          const TileElementIndex i_t{r + t_start.row(), t_start.col()};
+
+          DLAF_ASSERT_HEAVY(i_t.isIn(t.size()), i_t, t_size);
+          DLAF_ASSERT_HEAVY(i_v.isIn(vi.size()), i_v, vi.size());
+
+          t(i_t) = -tau * dlaf::conj(vi(i_v));
+        }
+
+        // skip alredy managed computations with implicit 1
+        va_start = va_start + TileElementSize{1, 0};
+        vb_start = vb_start + TileElementSize{1, 0};
+        va_size = {va_size.rows() - 1, va_size.cols()};
+      }
+
+      if (!va_size.isEmpty()) {
+        // clang-format off
+          blas::gemv(blas::Layout::ColMajor,
+              blas::Op::ConjTrans,
+              va_size.rows(), va_size.cols(),
+              -tau,
+              vi.ptr(va_start), vi.ld(),
+              vi.ptr(vb_start), 1,
+              1, t.ptr(t_start), 1);
+        // clang-format on
+      }
+    }
+    return std::move(t);
+  });
+  return hpx::dataflow(gemv_func, tile_vi, taus, tile_t);
+}
+
+template <class T>
+hpx::future<matrix::Tile<T, Device::CPU>> trmvUpdateColumn(
+    const TileElementIndex t_start, const TileElementSize t_size,
+    hpx::future<matrix::Tile<T, Device::CPU>>& tile_t) {
+  // Update each column (in order) t = T . t
+  // remember that T is upper triangular, so it is possible to use TRMV
+  auto trmv_func =
+      hpx::util::unwrapping([](auto&& tile_t, TileElementIndex t_start, TileElementSize t_size) {
+        // clang-format off
+      blas::trmv(blas::Layout::ColMajor,
+          blas::Uplo::Upper, blas::Op::NoTrans, blas::Diag::NonUnit,
+          t_size.rows(),
+          tile_t.ptr(), tile_t.ld(),
+          tile_t.ptr(t_start), 1);
+        // clang-format on
+
+        return std::move(tile_t);
+      });
+  return hpx::dataflow(trmv_func, tile_t, t_start, t_size);
+}
+
+template <class T>
+void QR_Tfactor<Backend::MC, Device::CPU, T>::call(const SizeType k, Matrix<const T, Device::CPU>& v,
+                                                   const LocalTileIndex v_start,
+                                                   hpx::shared_future<common::internal::vector<T>> taus,
+                                                   hpx::future<matrix::Tile<T, Device::CPU>> t) {
+  using hpx::util::unwrapping;
+  using common::make_data;
+
+  auto v_n = v.size().cols();
+  auto v_start_col = v_start.col();
+  auto v_nb = v.blockSize().cols();
+  auto v_col = v.nrTiles().cols();
+  // TODO: check
+  const SizeType panel_width = (v_start_col != v_col - 1) ? v_nb : v_n % v_nb;
```

Comment:
Why not simply:

https://github.com/eth-cscs/DLA-Future/blob/e9a51f5a0e87feb32e4f4f8f77b0f6e4a6ea9dcc/test/unit/solver/test_triangular.cpp#L152